universal-application-tool-0.0.1/test/UnitTest.java

```diff
@@ -101,4 +103,19 @@ public void testPostgresAdd() {
                         result ->
                             contentAsString(result).equals("person John with ID: 1234 added.")));
   }
+
+  // Unit test Postgres controller
+  @Test
+  public void testPostgresSyncAdd() {
+    HttpExecutionContext ec = new HttpExecutionContext(ForkJoinPool.commonPool());
+    PersonRepository repository = mock(PersonRepository.class);
+    Person person = mock(Person.class);
+    person.id = 1234L;
+    doNothing().when(person).save();
+    final PostgresController controller = spy(new PostgresController(repository, ec));
+    when(controller.newPerson("John")).thenReturn(person);
+    final Result result = controller.syncAdd("John");
+
+    contentAsString(result).equals("person John with ID: 1234 synchronously added.");
+  }
```

Comment:
I'm skeptical of the value of this unit test given its reliance on mocks (go/tott/289). The method is tested using a functional test which is adequate for our purposes imo.

Reply:
Removed. Thanks for the reference, that is very helpful.